watson/cli.py

```diff
@@ -773,6 +772,45 @@ def frames(watson):
     for frame in watson.frames:
         click.echo(style('short_id', frame.id))
 
+@cli.command(context_settings={'ignore_unknown_options': True})
+@click.argument('args', nargs=-1)
+@click.option('-f','--from', 'from_', required=True, type=Date, help="Date and time of start of tracked activity")
+@click.option('-t','--to', required=True, type=Date, help="Date and time of end of tracked activity")
+@click.pass_obj
+def add(watson, args, from_, to):
+    """
+    Add time for project with tag(s) that was not tracked live.
+    
+    Example:
+
+    \b
+    $ watson add --from "2018-03-20 12:00:00" --to "2018-03-20 13:00:00" programming +addfeature
+    """ 
+    # parse project name from args
+    project = ' '.join(
+        itertools.takewhile(lambda s: not s.startswith('+'), args)
+    )
+    # Find all the tags starting by a '+', even if there are spaces in them,
+    # then strip each tag and filter out the empty ones
+    tags = list(filter(None, map(operator.methodcaller('strip'), (
+        # We concatenate the word with the '+' to the following words
+        # not starting with a '+'
+        w[1:] + ' ' + ' '.join(itertools.takewhile(
+            lambda s: not s.startswith('+'), args[i + 1:]
+        ))
+        for i, w in enumerate(args) if w.startswith('+')
+    ))))  # pile of pancakes !
```

Comment:
we should write a (unit tested) utils for that :)

Reply:
I thought for a minute of separating that into its own function to avoid replicating code, but then I got lazy. But you are correct; separating that tag parsing into its own function and writing an unit test for that is a good idea.